cli/install_ee/install_ee.go

```diff
@@ -0,0 +1,163 @@
+package install_ee
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/tarantool/tt/cli/config"
+	"github.com/tarantool/tt/cli/util"
+	"github.com/tarantool/tt/cli/version"
+	"golang.org/x/term"
+)
+
+const eeSource string = "https://download.tarantool.io/enterprise/"
+
+type creds struct {
+	username string
+	password string
+}
+
+func getCredsInteractive() (creds, error) {
+	res := creds{}
+	reader := bufio.NewReader(os.Stdin)
+
+	fmt.Printf("Enter username: ")
+	resp, err := reader.ReadString('\n')
+	if err != nil {
+		return res, err
+	}
+	res.username = strings.TrimSpace(resp)
+
+	fmt.Printf("Enter password: ")
+	bytePass, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		return res, err
+	}
+	res.password = strings.TrimSpace(string(bytePass))
+
+	return res, nil
+}
+
+func getCredsFromFile(path string) (creds, error) {
+	res := creds{}
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return res, err
+	}
+
+	re := regexp.MustCompile("(?P<user>.*):(?P<pass>.*)")
+	matches := util.FindNamedMatches(re, strings.TrimSpace(string(data)))
+
+	if len(matches) == 0 {
+		return res, fmt.Errorf("Corrupted credentials.")
+	}
+
+	res.username = matches["user"]
+	res.password = matches["pass"]
+
+	return res, nil
+}
+
+func getCreds(cliOpts *config.CliOpts) (creds, error) {
+	if cliOpts.EE == nil || (cliOpts.EE != nil && cliOpts.EE.CredPath == "") {
+		return getCredsInteractive()
+	}
+
+	return getCredsFromFile(cliOpts.EE.CredPath)
+}
+
+func getTarballName(data string) (string, error) {
+	re := regexp.MustCompile(">(.*)<")
+
+	parsedData := re.FindStringSubmatch(data)
+	if len(parsedData) == 0 {
+		return "", fmt.Errorf("Cannot parse tarball name")
+	}
+
+	return parsedData[1], nil
+}
+
+// parseIndex collects a list of versions for the host architecture.
```

Comment:
What versions? Tarantool EE?
What is "data" parameter supposed to contain? Why is this called "parseIndex" and not "getAvailableVersions" or something like that?

Reply:
Renamed to getVersions.
Comment edited.